projects/iio_demo/src/app/main.c

```diff
@@ -194,12 +213,56 @@ int main(void)
 		.size = MAX_SIZE_BASE_ADDR
 	};
 
+	struct iio_data_buffer buff = {
+		.buff = (void *)ADC_DDR_BASEADDR,
+		.size = MAX_SIZE_BASE_ADDR
+	};
+
+	struct iio_data_buffer buff1 = {
+		.buff = (void *)DAC_DDR_BASEADDR,
+		.size = MAX_SIZE_BASE_ADDR
+	};
+
+	uint16_t** buf = (uint16_t**)malloc(10*sizeof(uint16_t*));
+
+	for(int i = 0; i < 500; i++) {
+		buf[i] = (uint16_t*)malloc(800*sizeof(uint16_t));
+	}
+
+	adc_init_par = (struct adc_demo_init_param) {
+		.loopback = buf,
+		.channel_no = 2,
+		.dev_global_attr = 3300,
+		.dev_ch_attr = 3311
+	};
+	status = adc_demo_init(&adc_desc, &adc_init_par);
+	if (status != 0)
```

Comment:
try to use error codes instead of raw values. For example `status != SUCCESS`. Check through the entire code.